lib/l10n_utils/fluent.py

```diff
@@ -0,0 +1,192 @@
+import json
+from functools import wraps
+from hashlib import md5
+
+from django.conf import settings
+from django.core.cache import caches
+from django.utils.encoding import force_bytes
+from django.utils.functional import lazy, cached_property
+
+from fluent.runtime import FluentLocalization, FluentResourceLoader
+from fluent.syntax.ast import GroupComment, Message
+
+from lib.l10n_utils import translation
+
+__all__ = [
+    'fluent_l10n',
+    'ftl',
+    'ftl_file_is_active',
+    'ftl_has_messages',
+    'ftl_lazy',
+    'get_metadata_file_path',
+    'has_messages',
+    'translate',
+]
+cache = caches['l10n']
+
+
+class FluentL10n(FluentLocalization):
+    def _localized_bundles(self):
+        for bundle in self._bundles():
+            if bundle.locales[0] == self.locales[0]:
+                yield bundle
```

Comment:
> This code shouldn't work for `en-US`, with most strings falling back to `en`, right?

This is only used for checking the percent translated and whether a translation has a message or not, so it's really only invoked on a single language instance of this class (the way I use it anyway). The `has_message` method always returns true for `en-*`, so it's working as expected.
 
> Also, it'd be more efficient to `break` on the first bundle that's not what we want, so that we don't need to exhaust the iterator, and thus, parse all locales.

Can you not have another language bundle from a different directory later in the list?

Reply:
> > This code shouldn't work for `en-US`, with most strings falling back to `en`, right?
> 
> This is only used for checking the percent translated and whether a translation has a message or not, so it's really only invoked on a single language instance of this class (the way I use it anyway). The `has_message` method always returns true for `en-*`, so it's working as expected.

Could this end up not wanting to activate en-US?

> > Also, it'd be more efficient to `break` on the first bundle that's not what we want, so that we don't need to exhaust the iterator, and thus, parse all locales.
> 
> Can you not have another language bundle from a different directory later in the list?

That'd be bad, I think. Technically, yes, it's possible, but in practice, we shouldn't show English strings if we have German ones, right? The built-in `_generate_bundles` strictly goes locale by locale.

Reply:
Right, but what if we had a `de/mozorg/mission.ftl` in our bedrock l10n folder, and there was also one in the strings from the l10n team's repo. That is a situation that is possible if the German marketing team wants to override a string but we leave the rest alone.